sys/cheri/cheric.h

```diff
@@ -145,6 +145,14 @@ cheri_is_address_inbounds(const void * __capability cap, ptraddr_t addr)
 	return (addr >= cheri_getbase(cap) && addr < cheri_gettop(cap));
 }
 
+/* Check if the capability is valid, unsealed and has the given permissions. */
+static inline bool
+cheri_is_unsealed_with_perms(const void * __capability cap, ptraddr_t perms)
```

Comment:
I wonder if we should call this something like `cheri_is_dereferencable` to better communicate the intended purpose, or maybe `cheri_can_dereference`?